src/live/src/user-to-user/stream.js

```diff
@@ -0,0 +1,245 @@
+   */
+  _unsubscribeFromSubstream(substreamOwnerId) {
+    const path = `${substreamOwnerId}/unsubscribe`;
+
+    return this._makeStreamRequest(path, RequestMethod.POST, { deviceId: this.client.deviceId })
+      .then((response) => {
+        delete this.subscribeChannels[substreamOwnerId];
+        return response;
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {MessageReceiver} receiver
+   * @returns {Promise}
+   */
+  _subscribe(userId, receiver) {
+    return this._requestSubscribeAccess(userId)
+      .then(() => {
+        const channelName = this.subscribeChannels[userId] || this._buildChannelName(userId);
+        return this.liveService.subscribeToChannel(channelName, receiver);
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @returns {Promise<void>}
+   */
+  _unsubscribe(userId) {
+    return this._unsubscribeFromSubstream(userId)
+      .then(() => {
+        const channelName = this.subscribeChannels[userId] || this._buildChannelName(userId);
+        this.liveService.unsubscribeFromChannel(channelName);
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  _publish(userId, message) {
+    if (this.publishChannels[userId]) {
+      return this._retriablePublish(userId, message);
+    }
+    return this._publishWithAccessRequest(userId, message);
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  _publishWithAccessRequest(userId, message) {
+    return this._requestPublishAccess(userId)
+      .then(() => {
+        const channelName = this.publishChannels[userId] || this._buildChannelName(userId);
+        return this.liveService.publishToChannel(channelName, message);
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  _retriablePublish(userId, message) {
+    const channelName = this.publishChannels[userId] || this._buildChannelName(userId);
+    return this.liveService.publishToChannel(channelName, message)
+      .catch((err) => {
+        let promise = Promise.reject(err);
+        if (err.statusCode === StatusCode.Forbidden) {
+          promise = this._publishWithAccessRequest(userId, message);
+        }
+        return promise;
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @returns {string} The name of the channel for that user
+   */
+  _buildChannelName(userId) {
+    return `${this.client.appKey}.s-${this.name}.u-${userId}`;
+  }
+
+  /**
+   * @private
+   * @param {string} substreamOwnerId
+   * @returns {Promise}
+   */
+  _requestPublishAccess(substreamOwnerId) {
+    return this._makeStreamRequest(`${substreamOwnerId}/publish`, RequestMethod.POST)
+      .then((resp) => {
+        if (resp.substreamChannelName) {
+          this.publishChannels[substreamOwnerId] = resp.substreamChannelName;
```

Comment:
This method overwrites it every time a call to `/publish` is made (to request access). It is called when the first `_publish()` attempt is made, or when a 403 is received when publishing to PubNub, so it will get overwritten/updated when necessary.